distribution/oceanbase-todo/internal/dsn.go

```diff
@@ -0,0 +1,40 @@
+package internal
+
+import (
+	"errors"
+	"os"
+	"strings"
+)
+
+type DSN struct {
+	Host     string
+	Port     string
+	User     string
+	Password string
+	Database string
+}
+
+func (d DSN) String() string {
+	return d.User + ":" + d.Password + "@tcp(" + d.Host + ":" + d.Port + ")/" + d.Database + "?charset=utf8mb4&parseTime=True&loc=Local"
+}
+
+func NewDSN(prefixes ...string) (*DSN, error) {
+	prefix := ""
+	if len(prefixes) > 0 {
+		prefix = strings.Join(prefixes, "_") + "_"
+	}
+	dsn := DSN{
+		Host:     os.Getenv(prefix + "DB_HOST"),
+		Port:     os.Getenv(prefix + "DB_PORT"),
+		User:     os.Getenv(prefix + "DB_USER"),
+		Password: os.Getenv(prefix + "DB_PASSWORD"),
+		Database: os.Getenv(prefix + "DB_DATABASE"),
+	}
+	if dsn.Port == "" {
```

Comment:
if the port is not provided, an error should be returned, DSN should be a low level model, and it's not necessary to add OceanBase related logic